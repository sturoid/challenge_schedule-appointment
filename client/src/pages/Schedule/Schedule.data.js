import React from 'react';
import gql from 'graphql-tag';
import { useQuery } from '@apollo/react-hooks';
import Schedule from './Schedule';
import Loading from '../../shared/Loading';
import ApiError from '../../shared/errors/ApiError';

const GET_APPOINTMENTS = gql`
  {
    coaches {
      id
      name
      appointments {
        id
        start(inTimezone: true)
        end(inTimezone: true)
      }
    }
  }
`;

const ScheduleData = () => {
  const { loading, error, data } = useQuery(GET_APPOINTMENTS);
  if (loading) return <Loading />;
  if (error) return <ApiError error={error} />;
  return <Schedule coaches={data.coaches} />;
};

export default ScheduleData;
