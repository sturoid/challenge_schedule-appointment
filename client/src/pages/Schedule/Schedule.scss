.schedule {
  display: flex;
}

.schedule-head {
  text-align: center;
  h1 {
    font-size: 4rem;
    font-weight: 100;
    margin: 0;
    padding: 20px 0 15px 0;
  }
  h2 {
    color: var(--grey700);
    font-style: italic;
    font-weight: 100;
    font-size: 1.7rem;
    margin: 0;
    padding: 0 0 30px 0;
  }
}
