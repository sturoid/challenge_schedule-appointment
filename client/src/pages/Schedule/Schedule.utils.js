export function getDateFromTimeSlot(timeSlot) {
  return new Date().setHours(timeSlot.hours, timeSlot.minutes, 0, 0);
}

export default getDateFromTimeSlot;
