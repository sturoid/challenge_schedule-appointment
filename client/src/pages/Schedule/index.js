export { default } from './Schedule.data';
