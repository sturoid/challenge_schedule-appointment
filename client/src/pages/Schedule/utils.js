export function getDateFromTimeSlot(timeSlot) {
  return new Date().setHours(timeSlot.hours, timeSlot.minutes, timeSlot.seconds);
}

export default getDateFromTimeSlot;
