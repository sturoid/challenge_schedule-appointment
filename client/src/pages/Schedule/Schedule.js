import React from 'react';
import PropTypes from 'prop-types';
import Page from '../../shared/Page';
import './Schedule.scss';

const Schedule = ({ coaches }) => {
  return (
    <Page>
      <h1>Schedule</h1>
      <section className="schedule">
        {coaches.map(({ id, name, appointments }) => {
          return (
            <div key={id} className="coach">
              <h2>{name}</h2>
              {appointments && appointments.length ? (
                <div className="appointments">
                  {appointments.map(a => {
                    return (
                      <div key={a.id}>
                        {a.start}:{a.end}
                      </div>
                    );
                  })}
                </div>
              ) : (
                'No appointments'
              )}
            </div>
          );
        })}
      </section>
    </Page>
  );
};

Schedule.props = {
  coaches: PropTypes.arrayOf(PropTypes.any)
};

export default Schedule;
