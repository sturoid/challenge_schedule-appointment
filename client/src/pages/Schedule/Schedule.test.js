import React from 'react';
import renderer from 'react-test-renderer';
import Schedule from './Schedule';

const coaches = [
  {
    id: 1,
    name: 'Bert Nelson',
    appointments: [
      {
        start: '2019-12-11 17:00:00Z',
        end: '2019-12-11 17:30:00Z'
      }
    ]
  },
  {
    id: 2,
    name: 'Bobby Booshay',
    appointments: [
      {
        start: '2019-12-11 17:00:00Z',
        end: '2019-12-11 17:30:00Z'
      },
      {
        start: '2019-12-11 18:30:00Z',
        end: '2019-12-11 19:00:00Z'
      }
    ]
  }
];

describe('Schedule', () => {
  it('renders correctly', () => {
    const com = renderer.create(<Schedule coaches={coaches} />).toJSON();
    expect(com).toMatchSnapshot();
  });
});
