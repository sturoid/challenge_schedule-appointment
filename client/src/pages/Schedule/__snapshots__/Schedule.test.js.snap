// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule renders correctly 1`] = `
<div
  className="page"
>
  <h1>
    Schedule
  </h1>
  <section
    className="schedule"
  >
    <div
      className="coach"
    >
      <h2>
        Bert Nelson
      </h2>
      <div
        className="appointments"
      >
        <div>
          2019-12-11 17:00:00Z
           - 
          2019-12-11 17:30:00Z
        </div>
      </div>
    </div>
    <div
      className="coach"
    >
      <h2>
        Bobby Booshay
      </h2>
      <div
        className="appointments"
      >
        <div>
          2019-12-11 17:00:00Z
           - 
          2019-12-11 17:30:00Z
        </div>
        <div>
          2019-12-11 18:30:00Z
           - 
          2019-12-11 19:00:00Z
        </div>
      </div>
    </div>
  </section>
</div>
`;
